jobs/job_test_utils.py

```diff
@@ -116,3 +162,83 @@ def assert_job_output_is(self, expected):
     def assert_job_output_is_empty(self):
         """Asserts that the output of self.JOB_CLASS is an empty PCollection."""
         self.assert_pcoll_empty(self.run_job())
+
+
+@contextlib2.contextmanager
+def decorate_beam_errors():
+    """Context manager to improve the readability of beam_testing_util errors.
+
+    The beam_testing_util module raises exceptions with a single string of
+    repr()'d lists as the message. The items end up appearing on one long line,
+    making it difficult to read when the elements of the lists are very long
+    (which they tend to be especially for Oppia's audit errors).
+
+    This context manager tries to split the list elements into lines, so that
+    it's easier to read which errors occurred and why. If it cannot parse the
+    message successfully, it will raise the same message.
+
+    Yields:
+        None. Nothing.
+
+    Raises:
+        AssertionError. The decorated exception.
+    """
+    try:
+        yield
+    except beam_testing_util.BeamAssertException as exception:
+        msg = python_utils.UNICODE(exception)
+        match = (
+            re.match(
+                r'.*'
+                r', unexpected elements (?P<unexpected>.*)'
+                r', missing elements (?P<missing>.*)'
+                r' \[(?P<context>while running .*)\]',
+                msg)
+            or re.match(
+                r'.*'
+                r', unexpected elements (?P<unexpected>.*)'
+                r' \[(?P<context>while running .*)\]',
+                msg)
+            or re.match(
+                r'.*'
+                r', missing elements (?P<missing>.*)'
+                r' \[(?P<context>while running .*)\]',
+                msg)
+            or re.match(
+                r'.*'
+                r'\[\] == (?P<unexpected>.*)'
+                r' \[(?P<context>while running .*)\]',
+                msg)
+        )
+
+        if match:
+            groupdict = match.groupdict()
+        else:
+            raise AssertionError(msg)
+
+        unexpected = groupdict.get('unexpected', None)
```

Comment:
I'm finding it a bit hard to parse this -- seems like there's a noun missing in the variable name; ditto below for "missing". Maybe elaborate a bit?

Reply:
Done.